pkg/daemon/route/utils.go

```diff
@@ -295,32 +302,106 @@ func ensureFromPodSubnetRuleAndRoutes(forwardNodeIfName string, cidr *net.IPNet,
 					return fmt.Errorf("set overlay route %v for table %v failed: %v", subnetRoute.String(), table, err)
 				}
 			}
+
+			// For overlay pod to access underlay excluded ip addresses, should not be forced to pass through vxlan device.
+			if err := ensureExcludedIPBlockRoutes(underlayExcludeIPBlockMap, table, family); err != nil {
+				return fmt.Errorf("ensure exclude all ip block routes failed: %v", err)
+			}
 		}
+
 	} else {
-		defaultRoute := &netlink.Route{
+		localAddrList, err := netlink.AddrList(nil, family)
+		if err != nil {
+			return fmt.Errorf("list local addresses failed: %v", err)
+		}
+
+		isLocalSubnet := false
+		for _, address := range localAddrList {
+			if cidr.Contains(address.IP) {
+				// Check if address is an enhanced address or used to connect a subnet.
+				if address.Flags&unix.IFA_F_NOPREFIXROUTE == 0 {
```

Comment:
if this judgment locate the local address except enhanced addresses? why?

Reply:
Enhanced addresses will always have the `IFA_F_NOPREFIXROUTE` flag.